### DISCUSSION TITLE
MicroG patches vs. GrapheneOS

### DISCUSSION THREAD

**Discussion by samopolacek** [ReVanced/revanced-discussions#787] [Question] [answered]
I might be misunderstanding something as I am new to GrapheneOS so please try to be gentle with me. 😄 

Caution, I might get some facts in the following text a bit wrong, or just plain straight retarded.

As you may know, root access is not available on stock GrapheneOS. That means that if one wants to patch, for example, YouTube app, MicroG patch has to be used, otherwise root permissions are required. When I tried to use the MicroG patch for YouTube, I thus managed to install the generated patched YouTube app. 
Launching this app, however, fails as MicroG is now required and is not on my phone, which makes sense. 
MicroG is not recommended on GrapheneOS due to security concerns. If I understand GrapheOS correctly, it implements its own middle-ware that provides all the implementation that MicroG provides.

Is there a way to make patches work on GrapheneOS without root and without MicroG?

**Comment by oSumAtrIX** [comment]
MicroG is open source, so i am unsure how its a security concern. If you do not want to use the seperate Vanced MicroG, you will need to exclude the MicroG patch and install the patched app under it's original package name which now is a concern of your OS.

**Comment by CnC-Robert** [comment]
Graphene uses normal play services and isolates it to give more privacy

Microg is an open source alternative to play services and is much better in terms of privacy and has no security concerns

Also yt revanced requires vanced microg instead of micro g which is a seperate user app

**Comment by raxod502** [comment, marked as answer]
Verified that installing https://github.com/TeamVanced/VancedMicroG apk on GrapheneOS and then patching YouTube Music via ReVanced Manager does indeed work perfectly.

**Comment by bam80** [comment]
Can I use both MicroG and isolated play services on Graphene OS?

**Reply by LisoUseInAIKyrios** [reply]
Yes you can use both.

ReVanced fork of microG uses a different package name, so you can install it on phones that have their own microG.

**Reply by Waymorehoss** [reply]
So like can i just use google play services and make the revanced app work or do i still need microG to make it work?

**Reply by oSumAtrIX** [reply]
try and see
